api/src/main/java/org/pmiops/workbench/db/dao/DataSetServiceImpl.java

```diff
@@ -534,6 +537,28 @@ private String getQualifiedColumnName(Domain currentDomain, String columnName) {
         .collect(Collectors.toList());
   }
 
+  @Override
+  @Transactional
+  public DataSet cloneDataSetToWorkspace(
+      DataSet fromDataSet, Workspace toWorkspace, Set<Long> cohortIds, Set<Long> conceptSetIds) {
+    DataSet toDataSet = new DataSet(fromDataSet);
+    toDataSet.setWorkspaceId(toWorkspace.getWorkspaceId());
+    toDataSet.setCreatorId(toWorkspace.getCreator().getUserId());
+    toDataSet.setLastModifiedTime(toWorkspace.getLastModifiedTime());
```

Comment:
Why are you using the destination workspace's modified time instead of the source dataset's? Same question for creation time.

Reply:
Yup! I thought about recreating `now` here, but decided to follow the existing pattern.

Reply:
I think it would be clearer to declare a `now` explicitly and use it. As written, it's definitely suggesting we want the workspace's times used for the dataset's times, and looks like a bug. So maybe a comment and/or declaration that this is deliberate (and why it's desired) would help.

It's also confusing that the new dataset's creator is the same as the old *workspace's* creator. In the case where the fromDataset was itself cloned into the fromWorkspace, this may well not be correct. But I guess we need to define (and document) the semantics: is the cloning user the creator, or the original dataset's creator, or the original workspace's creator?

Also, I'd consider the case where I take a year-old dataset and clone it to a new workspace (or an even older workspace). I certainly wouldn't expect a cloned workspace to have timestamps older than the creation time of the source dataset.